PurchasesCoreSwift/Networking/Backend.swift

```diff
@@ -0,0 +1,650 @@
+        }
+    }
+
+    @objc public func logIn(currentAppUserID: String,
+                            newAppUserID: String,
+                            completion: @escaping (PurchaserInfo?, Bool, Error?) -> Void) {
+
+        let requestBody = ["app_user_id": currentAppUserID, "new_app_user_id": newAppUserID]
+        httpClient.performPOSTRequest(serially: true,
+                                      path: "/subscribers/identify",
+                                      requestBody: requestBody,
+                                      headers: authHeaders) { statusCode, response, error in
+
+            self.handleLogin(response: response, statusCode: statusCode, error: error, completion: completion)
+        }
+    }
+
+    @objc(getOfferingsForAppUserID:completion:)
+    public func getOfferings(appUserID: String, completion: @escaping OfferingsResponseHandler) {
+        guard let appUserID = try? escapedAppUserID(appUserID: appUserID) else {
+            completion(nil, ErrorUtils.missingAppUserIDError())
+            return
+        }
+
+        let path = "/subscribers/\(appUserID)/offerings"
+        if add(callback: completion, key: path) {
+            return
+        }
+
+        httpClient.performGETRequest(serially: false, path: path, headers: authHeaders) { statusCode, response, error in
+            if error == nil && statusCode < HTTPStatusCodes.redirect.rawValue {
+                for callback in self.getOfferingsCallbacksAndClearCache(forKey: path) {
+                    callback(response, nil)
+                }
+                return
+            }
+
+            let errorForCallbacks: Error
+            if let error = error {
+                errorForCallbacks = ErrorUtils.networkError(withUnderlyingError: error)
+            } else if statusCode > HTTPStatusCodes.redirect.rawValue {
+                let backendCode = self.maybeNumberFromError(code: response?["code"])
+                let backendMessage = response?["message"] as? String
+                errorForCallbacks = ErrorUtils.backendError(withBackendCode: backendCode,
+                                                            backendMessage: backendMessage)
+            } else {
+                errorForCallbacks = ErrorUtils.unexpectedBackendResponseError()
+            }
+
+            for callback in self.getOfferingsCallbacksAndClearCache(forKey: path) {
+                callback(nil, errorForCallbacks)
+            }
+        }
+    }
+
+    @objc(getIntroEligibilityForAppUserID:receiptData:productIdentifiers:completion:)
+    public func getIntroEligibility(appUserID: String,
+                                    receiptData: Data,
+                                    productIdentifiers: [String],
+                                    completion: @escaping IntroEligibilityResponseHandler) {
+        guard productIdentifiers.count > 0 else {
+            completion([:], nil)
+            return
+        }
+
+        if receiptData.count == 0 {
```

Comment:
nitpick: `receiptData.isEmpty()`

Reply:
🤔 kinda feels like this bit is out of scope for the Backend class, so whenever we do https://github.com/RevenueCat/purchases-ios/issues/690 we should probably move this over. 
I'll add a note there

Reply:
Yeah, thanks!